### DISCUSSION TITLE
MP4 or MPEG stream?

### DISCUSSION THREAD

**Discussion by R-and-all** [thirtythreeforty/neolink#347] [Ideas]
Hi 

Im tryign to setup a little dashboard for Hubitat, and found your software after googling reolink hack.
After settign everything up I now have access to my cameras via neolink (thank you so much). 
However my next task is to get a small thumbnail or player into hubitat. 
After looking for a quick rtsp plugin for hubitat, I spotted 
https://community.hubitat.com/t/access-ip-camera-using-rstp-url/20420/2 
which seems to imply I need mp4 or mpeg stream. Is this something neolink could do? so rather than starting up as 
neolink rtsp --config <path> 
could you choose the stream type?

I'm going to start things off with vlc doing it for me for now, or tinycam, I still need to decide, but this would be a great feature to have, for me!

**Comment by QuantumEntangledAndy** [comment]
This is possible but not sures there's much call for it. I'm surprised that hubitat dosent have a generic rtsp plugin like home assistant does though.

I'm also not sure if converting to mp4 is really that way. It still won't likely understand the rtsp protocol itself. Browser just don't know how to talk with `rtsp://` links regardless of the format underneath.

What homeassistant does in their generic rtsp plugin is dump the rtsp stream into a jpeg stream and then pushes that into its webpages. Maybe that would be the way to tackle this.

**Reply by R-and-all** [reply]
ok thanks for the reply.

Im going to ask here and search for the answer. 
If I have cameras setup in neolink are they always "on" as in are they always being polled for the image, I have the argus line on battery so wondered if I needed to hard wire a usb power brick to get them working as I need, or if neolink only looks and activates the streaming when I choose to look at teh camera?

**Reply by R-and-all** [reply]
Oh the other thing, is can you set the camera to just grab a static image and show that every n seconds? 
I wanted a way to browse teh camera without opening a stream. so if it refreshed its self every n seconds and had that image somewhere, then when opening the stream it was live that would be amazing.

**Reply by QuantumEntangledAndy** [reply]
The thirtythreeforty fork is always on. Even when not connected with an rtsp client. My fork does have some WIP changes to address this but it's a bit temperamental. I'm working on a better way to detect a client in gatreamer atm but will take a little while to finish.

There is an image command in my fork too. I already have someone who uses it to fetch a jpeg every few seconds. Maybe have a look at that.
